packages/cw22/src/lib.rs

```diff
@@ -0,0 +1,168 @@
+/*!
+The standard used to declare which interface contract implements
+This standard is inspired by the EIP-165 from Ethereum.
+
+For more information on this specification, please check out the
+[README](https://github.com/CosmWasm/cw-plus/blob/main/packages/cw22/README.md).
+ */
+
+mod query;
+
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::StdError;
+use cosmwasm_std::{StdResult, Storage};
+use cw_storage_plus::Map;
+use query::VersionResponse;
+use semver::{Version, VersionReq};
+
+pub const SUPPORTED_INTERFACES: Map<String, String> = Map::new("supported_interfaces");
+
+#[cw_serde]
+pub struct ContractSupportedInterface {
+    /// supported_interface is an optional parameter returning a vector of string represents interfaces
+    /// that the contract support The string value is the interface crate names in Rust crate Registry.
+    /// This parameter is inspired by the EIP-165 from Ethereum.
+    /// Each string value should follow a common standard such as <Registry Domain>:<Crate Name>
+    /// e.g "crates.io:cw2"
+    /// NOTE: this is just a hint for the caller to adapt on how to interact with this contract.
+    /// There is no guarantee that the contract actually implement these interfaces.
+    pub supported_interface: String,
+    /// semantic version on release tags.
+    /// e.g  "v0.16.0"
+    pub version: String,
+}
+
+/// set_contract_supported_interface should be used in instantiate to store the original version
+pub fn set_contract_supported_interface(
+    store: &mut dyn Storage,
+    mut supported_interfaces: Vec<ContractSupportedInterface>,
+) -> Result<(), StdError> {
+    while let Some(supported_interface) = supported_interfaces.pop() {
```

Comment:
can we use `for supported_interface in supported_interfaces {}` here?

Reply:
yes, the code will be similar.

Reply:
Argument change would force this for loop here, which is better anyway